proctor-common/src/main/java/com/indeed/proctor/common/ForceGroupsOptionsStrings.java

```diff
@@ -17,60 +24,238 @@
+     * The format of the payloadString is the following:
+     * <ul>
+     *     <li>payloadType:payloadValue</li>
+     *     <li>Where payloadType is one of the 7 payload types supported by Proctor (stringValue, stringArray, doubleValue, longValue, longArray, map)</li>
+     *     <li>If payloadValue is an array is expected in the following format: [value,value,value]</li>
+     *     <li>If payloadValue is a map expecting following format: [key:value,key:value,key:value]</li>
+     * </ul>
+     */
+    @Nullable
+    public static Payload parseForcePayloadString(final String payloadString )
+    {
+        final Payload payload = new Payload();
+
+        final String[] payloadPieces = StringEscapeUtils.escapeJava(payloadString).split(":",2);
+
+        try {
+            final PayloadType payloadType = PayloadType.payloadTypeForName(payloadPieces[0]);
+            final String payloadValue = payloadPieces[1];
+            switch (payloadType) {
+                case DOUBLE_VALUE:
+                {
+                    payload.setDoubleValue(Double.parseDouble(payloadValue));
+                    break;
+                }
+                case DOUBLE_ARRAY:
+                {
+                    payload.setDoubleArray(Arrays.stream(getPayloadArray(payloadValue))
+                            .map(Double::valueOf)
+                            .toArray(Double[]::new) );
+                    break;
+                }
+                case LONG_VALUE:
+                {
+                    payload.setLongValue(Long.parseLong(payloadValue));
                     break;
                 }
+                case LONG_ARRAY:
+                {
+                    payload.setLongArray(Arrays.stream(getPayloadArray(payloadValue))
+                            .map(Long::valueOf)
+                            .toArray(Long[]::new));
+                    break;
+                }
+                case STRING_VALUE:
+                {
+                    // Remove string escaped quotes ("\)
+                    payload.setStringValue(payloadValue.substring(2,payloadValue.length()-2));
+                    break;
+                }
+                case STRING_ARRAY:
+                {
+                    // Remove outside brackets e.g. ([abc, def, ...])
+                    payload.setStringArray(getPayloadStringArray(payloadValue.substring(1,payloadValue.length()-1)));
+                    break;
+                }
+                case MAP:
+                {
+                    // Remove outside brackets e.g. (map:[keys:values, ...])
+                    payload.setMap(getPayloadMap(payloadValue.substring(1,payloadValue.length()-1)));
+                    break;
+                }
+            }
+        }
+        catch (final IllegalArgumentException | ArrayStoreException | ClassCastException e) {
+            return null;
+        }
+
+        return payload;
+    }
+
+    public static String[] getPayloadArray(final String payloadValue) {
```

Comment:
What's the difference between `getPayloadArray` and `getPayloadStringArray`?

Reply:
Get payload array is used for parsing arrays that are Doubles or Longs. Get Payload String requires extra parsing as users could put commas inside of the string within the arrays.